### DISCUSSION TITLE
Dificuldade em achar emprego

### DISCUSSION THREAD

**Discussion by helntk** [frontendbr/forum#1466] [Discussão]
Vou ser bem conciso no meu texto. Faz 6 meses que estou estudando FrontEnd com disciplina e tenho tido bons resultados. 
Aprendi: Html, Css, Bootstrap, Javascript, Jquery, Github e React(webpack o básico). Porém tenho tido dificuldades em saber o que estudar agora para conseguir entrar no mercado. Será que meu conhecimento é suficiente para começar? Devo aprender mais coisas antes de sair procurando vagas?

Meu site com alguns projetos: helcio.netlify.com

**Comment by vsppedro** [comment]
Não desista! Continue tentando e estudando! Pelo que eu vi, está no caminho certo.

Não sei se é o seu caso, se não for ignore, mas é o meu e por isso vou dividir uma coisa que eu já  repeti para mim várias: nunca se deixe intimidar pelo que é requisitado na vaga. Seja experiência ou tecnologias. Se achar uma vaga que quer muito, aplique!  Quem sabe mostrando o seu entusiamo e vontade de aprender o recrutador não lhe dar uma chance de seguir no processo seletivo? Isso acontece! Se ocorrer a temida rejeição, faz parte, mas saímos desse processo mais forte, pois com um possível feedback sabemos o que buscar para melhorar.

Um [artigo](https://www.themuse.com/advice/actually-you-should-apply-for-jobs-youre-not-100-qualified-for) interessante que acabei lendo um pouco antes de encontrar essa Issue.

PS: vou ficar acompanhando o desenrolar dessa história. Boa sorte, ou melhor; sucesso!

**Comment by gabriel-hahn** [comment]
Pelo que vi, tu está no caminho correto. Uma dica seria se aprofundar nessas tecnologias que tu citou, pois em 6 meses estudando tudo isso, é um pouco complicado assimilar todos conceitos de HTML, CSS, JavaScript e um framework (React no seu caso) a fundo para alguém que está iniciando. Mantém o foco nessas tecnologias e tenta entender elas, fazendo mais projetos e etc. Já é mais que o suficiente para conseguir um emprego como front-end em um cargo iniciante.
Quando iniciei a trabalhar como desenvolvedor profissionalmente, acredito que tinha menos conhecimento do que tu tens agora. Não desiste e já começar a aplicar para as vagas da tua região. Boa sorte :)

**Comment by helntk** [comment]
> Um [artigo](https://www.themuse.com/advice/actually-you-should-apply-for-jobs-youre-not-100-qualified-for) interessante que acabei lendo um pouco antes de encontrar essa Issue.
> 
Gostei desse artigo, acho que vai ajudar bastante nas futuras entrevistas, pelo jeito parece que eu tenho sido um pouco robótico ou frio. Talvez falte mesmo mostrar a vontade e interesse. Vai ser díficil por eu ser tímido, mas não tenho dificuldade em falar.

**Comment by helntk** [comment]
>pois em 6 meses estudando tudo isso, é um pouco complicado assimilar todos conceitos de HTML, CSS, JavaScript
>
Modéstia a parte, minha base de Javascript é muito boa, tenho lido bastante livros e lido documentações, feito vídeo, acompanho projetos no Youtube, etc. 
Acho que o que falta é um maior marketing da minha parte em vender, o produto. Ou talvez alguma técnologia que eu esteja esquecendo.

**Comment by felipefialho** [comment]
>  Devo aprender mais coisas antes de sair procurando vagas?

Na verdade é quase o contrário, diria para aprender menos coisas e focar em necessidades especificas de vagas que te interessem.

Uma sugestão é olhar o [frontend-challenges](https://github.com/felipefialho/frontend-challenges), dá pra ter uma boa noção da exigência atual do mercado.

**Comment by helntk** [comment]
> Na verdade é quase o contrário, diria para aprender menos coisas e focar em necessidades especificas de vagas que te interessem.
>
Entendo. Então, pra uma entrada no mercado seria interessante saber bem os fundamentos e não ficar indo muito avançado em complexidade?

**Comment by ViniciusGularte** [comment]
Contando  o que eu fiz, no caso queria trabalhar muito com react, peguei  durante 1 ano inteiro só estudando react, redux, redux thunk, toda arquitetura flux e muito Javascript,  1 ano inteiro pra +, além disso fui juntando grana para poder me mudar de cidade durante todo este tempo, nesse período  ficava me candidatando pra outras vagas só pra me testar, meu psicológico e ver como os recrutadores tavão fazendo perguntas/desafios de código(Repositorio do @felipefialho  tem muitos desafios legais),  eu via outras coisas com vue, php, node, angular, e outros, porém ignorava estas coisas e fiquei focando só no que importava, no que eu teria que masterizar pra conseguir uma vaga boa/foda na área que eu queria, ficava vasculhando  o [frontend-vagas](https://github.com/frontendbr/vagas/issues) pra saber as tecnologias que o pessoal tava trabalhando, depois quando decidi que era a hora certa, comecei a me candidatar fulltime, fazia 5 entrevistas por semana kkk, muitas entrevistas sem feedback, muitas com um não, mas uma que divulgaram em um grupo no slack deu certo, hoje estou numa empresa que gosto muito, sai da manutenção de um sistema monolítico Php pra criar a arquitetura do novo Spa da empresa que trabalho atualmente, minha dica é dominar o básico bem e pegar algum framework que vc mais se identifica pra masterizar, claro que pensando no que o mercado pede também, depois que você conseguir entrar só vai, boa sorte nos estudos :smile: :rocket: 

Você esta no caminho certo :wink:

**Comment by fernandoRj** [comment]
tava pensando aqui, o mercado de trabalho para deves front-end e mt concorrido, vc dispulta vagas com pessoas que ja tem experiencia as empresas so querem funcionarios experientes e que ja estao produzindo e gerando lucro para as empresas no mercado de ti uma pessoa sem experiencia e a msm coisa que nada, diferente de outras areas como administraçao e direito, onde vc ja pode exercer a profissao e produzir para a empresa que vc trabalha msm sem experiencia

**Comment by gabriel-hahn** [comment]
Não se pode generalizar. Existem muitas empresas em busca de profissionais em nível iniciante também, justamente por ser um mercado concorrido e ter uma escassez de profissionais disponíveis. Quase todo profissional "sênior" já foi um desenvolvedor júnior em alguma empresa um dia. 
O que é necessário para iniciantes na minha opinião é entender um pouco melhor a importância dos fundamentos da linguagem e focar no que realmente é necessário conforme o @felipefialho já comentou.
Olhar as vagas, ver o que o mercado está exigindo, ver qual tecnologia tu mais gosta e estudar mantendo o foco.

**Comment by tcelestino** [comment]
Uma dica: não só estude, faça algo e mostre para o "mundo"! Que seja um boneco feito com CSS, hoj em dia é bem mais fácil você ser encontrado por empresas de todos os tipos se você mostrar o seu trabalho. Antigamente, as empresas buscavam seu portfolio (quem nunca teve um .com/.com.br? :smile:), hoje em dia elas estão de olho no que você já fez independente dos lugares que passou/estudou.

Uma "armadilha" que vejo para as pessoas que estão começando a estudar "front-end" é que mercado tem passado a sensação que será mais fácil de encontrar um trabalho porque a demanda esa alta, mas isso é uma bolha e todo bolha você precisa ter cuidado. Pode parecer ser uma maravilha você trabalhar com essas tecnologias do momento, mas o que vai contar mesmo para qualquer empresa não é apenas sua habilidade em js, css, etc... mas sim seu comportamento, sua mentalidade e sua pro-atividade e o mais importante de todos: o que você vai entregar para a empresa! 

No mais, continue na sua busca, uma hora vai encontrar algo! 

:crab:

**Comment by helntk** [comment]
>Uma dica: não só estude, faça algo e mostre para o "mundo"!
>
Tenho esse meu portfólio, mas acho que você tem razão. Preciso de algo criativo, tava pensando em criar uma rede social em react e uma loja de livros pra mostrar.

**Comment by theandersonn** [comment]
Meu amigo, uma dica!

Dê um capricho no seu perfil  aqui no github, no momento não tem o seu nome, não tem foto, descrições dos projetos... isso passa má impressão.

Nosso perfil no github e linkedin devem transparecer o desenvolvedor que somos é e estamos nos tornando a cada dia. Em cada projeto, no readme, descreva o porque você fez, o que aprendeu, porque foi dificil... quando um desenvolvedor mais experiente ler essas descrições e o seu código, vai conseguir ver a sua evolução.

**Comment by felipefialho** [comment]
Perfeito @theandersonn... perfeito.

**Comment by jair-anderson-souza** [comment]
@helntk, você já deu uma olhada em plataformas semelhantes ao BossaBox? Pode ser uma alternativa interessante tbm pra você.

**Comment by helntk** [comment]
>
Em cada projeto, no readme, descreva o porque você fez, o que aprendeu, porque foi dificil... quando um desenvolvedor mais experiente ler essas descrições e o seu código, vai conseguir ver a sua evolução.
>
Obrigado cara, todas as dicas foram boas e estou usando de guia. Mas essa foi uma bem mais prática. A gente que tá começando não tem muita noção, por isso é importante essa troca.

**Comment by helntk** [comment]
> @helntk, você já deu uma olhada em plataformas semelhantes ao BossaBox? Pode ser uma alternativa interessante tbm pra você.

Não, vou dar uma olhada. Obrigado!

**Comment by ninetails** [comment]
só pra cobrar o @helntk: bota fotinho, bota nome, capricha no LinkedIn rs

E todo projeto, mesmo que de brincadeira ou aprendizado de código, joga no GitHub. Pelo menos se algum dia vc precisar de referência, vai lembrar que provavelmente tá lá de backup. Por isso a importância de fazer um bom README.md e descrição no repo pra vc se achar depois.

**Comment by helntk** [comment]
> só pra cobrar o @helntk: bota fotinho, bota nome, capricha no LinkedIn rs
> 


Obrigado. Criei o linkedin, já estou melhorando esse lado. Já fui chamado para uma entrevista.
